### PR TITLE
Port to macOS.

### PR DESCRIPTION
**Pull request by zoujiaqing**
Commit without test.

### CODE REVIEW

**Review by frannecki — COMMENTED** [review]
Failed to compile on linux
> [ 1%] Building CXX object CMakeFiles/ladder.dir/ladder/src/Acceptor.cpp.o
[ 2%] Building CXX object CMakeFiles/ladder.dir/ladder/src/ArgumentParser.cpp.o
[ 4%] Building CXX object CMakeFiles/ladder.dir/ladder/src/Base.cpp.o
[ 5%] Building CXX object CMakeFiles/ladder.dir/ladder/src/Buffer.cpp.o
[ 7%] Building CXX object CMakeFiles/ladder.dir/ladder/src/Channel.cpp.o
[ 8%] Building CXX object CMakeFiles/ladder.dir/ladder/src/Connection.cpp.o
[ 10%] Building CXX object CMakeFiles/ladder.dir/ladder/src/EventLoop.cpp.o
[ 11%] Building CXX object CMakeFiles/ladder.dir/ladder/src/EventLoopThreadPool.cpp.o
[ 12%] Building CXX object CMakeFiles/ladder.dir/ladder/src/EventPoller.cpp.o
/mnt/ladder/ladder/src/EventPoller.cpp: In constructor ‘ladder::Pipe::Pipe()’:
/mnt/ladder/ladder/src/EventPoller.cpp:229:33: error: ‘pipe_’ was not declared in this scope; did you mean ‘pipe2’?
229 | const auto r = ::pipe2(&pipe_[0], O_CLOEXEC);
| ^~~~~
| pipe2
/mnt/ladder/ladder/src/EventPoller.cpp:231:19: error: ‘exception’ was not declared in this scope; did you mean ‘std::exception’?
231 | throw exception("pipe2 failed: " +
| ^~~~~~~~~
| std::exception
In file included from /usr/include/c++/11/new:41,
from /usr/include/c++/11/functional:53,
from /mnt/ladder/ladder/include/Channel.h:14,
from /mnt/ladder/ladder/src/EventPoller.cpp:12:
/usr/include/c++/11/bits/exception.h:60:9: note: ‘std::exception’ declared here
60 | class exception
| ^~~~~~~~~
make[2]: *** [CMakeFiles/ladder.dir/build.make:188: CMakeFiles/ladder.dir/ladder/src/EventPoller.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:166: CMakeFiles/ladder.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

File: ladder/src/EventPoller.cpp [hunk 1]
```diff
@@ -222,9 +222,26 @@ void EventPoller::SetWakeupCallback(const std::function<void()>& callback) {
 
 #ifdef LADDER_OS_UNIX
 Pipe::Pipe() {
+#ifndef LADDER_OS_MAC
   if (::pipe2(fd_, O_NONBLOCK) < 0) {
     EXIT("pipe2");
   }
+        const auto r = ::pipe2(&pipe_[0], O_CLOEXEC);
+        if (-1 == r)
+            throw exception("pipe2 failed: " +
```
Comment (frannecki) [inline ladder/src/EventPoller.cpp:231]: Please do not use `std::exception`. Instead use `EXIT` from `ladder/include/utils.h` to keep consistent with the repo

**Review by frannecki — COMMENTED** [review]
Thank you! I'll be testing its functionality before approval.

**Comment by zoujiaqing** [comment]
OK, I tested ladder on macOS.

**Review by frannecki — APPROVED** [review]

**Pull request merged**